### PR TITLE
What is Production?

### PR DESCRIPTION
**Pull request by mikekaminsky**

### CODE REVIEW

**Comment by mikekaminsky** [comment]
Helpful comment, @ilanman! 

I think I'd quibble with this line
> analysts shouldn't touch the customer-facing app (prod) because

Because I believe that analysts (broadly defined to include data scientists) *should* touch the customer-facing production application, we just need to get better at building applications and infrastructure in such a way that this is possible (similar to how we have developed tools and interfaces that allow front-end engineers to do front-end stuff and back-end engineers to do back-end stuff without either getting in each other's way).

Additionally, depending on your business, your email-sending logic could be just as complex (if not more!) than your customer-facing application (this has been basically true at some of my past jobs).

I'm going to re-work the post to better highlight these areas of both disagreement and agreement which I think this will help make the post more clear -- even if we don't agree on everything :)

**Comment by ilanman** [comment]
Cool. Just to be clear, my bullet:

> analysts shouldn't touch the customer-facing app (prod) because

is just reiterating this sentiment:

> I have spoken to many fellow analytics practitioners who are either adament that they want their team to *never* touch "production."

I was just laying out a logical flow, rather than stating my beliefs. 

In fact, I agree that analysts should be able to deploy to production given an environment that is well structured for their type of work (e.g., updating weights in a `.json` or perhaps some sort of Input (X) -> function (Logistic regression) -> Output (y) paradigm)

**Comment by mikekaminsky** [comment]
Ah, got it! Thanks @ilanman. 

Apologies!

File: content/post/what-is-production.md [hunk 1]
```diff
@@ -10,51 +10,58 @@ tags:
...
 
 * Local development
 * Remote development
 * Staging / QA
 * Production
 
-Every “application” will have multiple environments, one of which will be the one facing non-developer users. The one facing non-developer users is typically called “production”. Before we get too into the weeds, I want to start with my boldest claim: all of your analysts are deploying to production every day, already. Let me walk you through my favorite[^1] line of questioning:
+Every “application” will have multiple environments, one of which will be the one facing non-developer users. The one facing non-developer users is typically called “production”. 
+
+Here's my thesis: It's critical for Analytics teams to impact customer facing applications. If it is not possible to enable that in a safe way, then you should modify your customer facing application in such a way to make that possible.
```
Comment (ilanman) [inline content/post/what-is-production.md]: Do you want to add a qualifier such as "If you believe that Analytics should play a role in your customer facing app"? I can imagine organizations that have customer facing apps that aren't data-driven in any meaningful way.

But okay to ignore this if you think it muddies the water or you address it further on, which I think you might.

File: content/post/what-is-production.md [hunk 2]
```diff
@@ -66,3 +73,4 @@ Now that we've discussed why this might be a good idea, we can talk about some w
 I've written about the [code-as-configuration]({{< relref "code-as-configuration.md" >}}) design pattern before, and in general this is a great pattern for empowering non-engineers to savely deploy code to production environments. Friend-of-the-blog Harlan Harris has also written about [using domain specific languages](http://proceedings.mlr.press/v82/harris18a/harris18a.pdf) to facilitate testing, coordinating, and deploying machine learning models in production.
 
 [^1]: and most obnoxious. People love me at parties.
+[^1]: As usual, In this post I use the term "Analytics" and "analysts" in the most broad way possible including data scientists, data analysts and other data experts.
```
Comment (ilanman) [inline content/post/what-is-production.md]: `in this`

File: content/post/what-is-production.md (hunk 1 above)
Comment (mikekaminsky) [reply content/post/what-is-production.md]: Good call. I'm going to drop a footnote.

**Pull request merged**